src/core/dxf/qgsdxfexport.cpp

```diff
@@ -1682,39 +1667,28 @@ void QgsDxfExport::addFeature( QgsSymbolRenderContext &ctx, const QgsCoordinateT
         }
 
         const QgsCurvePolygon *polygon = dynamic_cast<const QgsCurvePolygon *>( sourceGeom );
-        Q_ASSERT( polygon );
-
-        writePolyline( *polygon->exteriorRing(), layer, lineStyleName, penColor, width );
-        for ( int i = 0; i < polygon->numInteriorRings(); i++ )
-          writePolyline( *polygon->interiorRing( i ), layer, lineStyleName, penColor, width );
-
-        break;
-      }
-
-      case QgsWkbTypes::MultiSurface:
-      case QgsWkbTypes::MultiPolygon:
-      {
-        if ( !qgsDoubleNear( offset, 0.0 ) )
+        if ( polygon )
         {
-          QgsGeos geos( sourceGeom );
-          tempGeom.reset( geos.buffer( offset, 0, Qgis::EndCapStyle::Flat, Qgis::JoinStyle::Miter, 2.0 ) );  //#spellok
-          if ( tempGeom )
-            sourceGeom = tempGeom.get();
-          else
-            sourceGeom = geom.constGet();
+          writePolyline( *polygon->exteriorRing(), layer, lineStyleName, penColor, width );
+          for ( int i = 0; i < polygon->numInteriorRings(); i++ )
+            writePolyline( *polygon->interiorRing( i ), layer, lineStyleName, penColor, width );
         }
-
-        const QgsGeometryCollection *gc = dynamic_cast<const QgsGeometryCollection *>( sourceGeom );
-        Q_ASSERT( gc );
-
-        for ( int i = 0; i < gc->numGeometries(); i++ )
+        else
         {
-          const QgsCurvePolygon *polygon = dynamic_cast<const QgsCurvePolygon *>( gc->geometryN( i ) );
-          Q_ASSERT( polygon );
+          const QgsGeometryCollection *gc = dynamic_cast<const QgsGeometryCollection *>( sourceGeom );
+          Q_ASSERT( gc );
+          if ( gc )
```

Comment:
So ... now I'm wondering if we need this check at all, at this point we *should* be sure we have a gc

Reply:
> at this point we should be sure we have a gc

I think so, but can't swear :-)  The current code should be safe enough: not crashing in production builds, and asserting in debug builds so we can investigate what's going on

Reply:
Makes sense. I always lived by the rule after an assert ub is allowed in prod. But not sure that's a common understanding 🤪